src/FullBuild/Baselines.fs

```diff
@@ -61,84 +62,131 @@ with
+                    |> Set.ofSeq
+        Baseline(buildInfo, bookmarks)    
+        
+    let serializeBaseline (baseline:Baseline) =
+        let repositories = 
+            baseline.Bookmarks 
+            |> Seq.map(fun b -> let r = BaselineFile.repositories_Item_Type()
+                                r.name <- b.Repository.Name
+                                r.version <- b.Version
+                                r)
+            |> Array.ofSeq
+        let baselineFile = BaselineFile()
+        baselineFile.branch <- baseline.Info.Branch
+        baselineFile.buildnumber <- baseline.Info.BuildNumber
+        baselineFile.repositories <- repositories
+        baselineFile
+
+    let getClonedReposBookmarks () =
+        let getRepoHash wsDir (repo : Repository) = async {
+            let changesetId = Tools.Vcs.LastCommit wsDir repo ""
+            return Bookmark(repo, changesetId)
+        }
+        graph.Repositories 
+        |> Set.add graph.MasterRepository
+        |> Seq.filter (fun x -> x.IsCloned)
+        |> Threading.ParExec (getRepoHash wsDir)
+        |> Set.ofSeq
+    
+    let updateBookmarks (source:Bookmark set) (overrideWith:Bookmark set) =
+        let dif = 
+            source 
+            |> Seq.where(fun b -> overrideWith 
+                                    |> Seq.exists (fun b1 -> b1.Repository.Name = b.Repository.Name) 
+                                    |> not) 
+            |> Set.ofSeq
+        Set.union dif overrideWith
+
+    member this.GetPulledBaseline () : Baseline option =
+        let baselineFileInfo = Env.GetBaselineFile() 
+        if baselineFileInfo.Exists then
+            baselineFileInfo.FullName |> parseBaselineFile |> Some
+        else
+            None
+
+    member this.GetSourcesBaseline () : Baseline =
+        let pulledBaseline = this.GetPulledBaseline()
+        let sourcesBookmarks = getClonedReposBookmarks ()
+
+        let resultBaseline = 
+            match pulledBaseline with
+            | Some baseline -> updateBookmarks baseline.Bookmarks sourcesBookmarks
+            | None -> sourcesBookmarks
+
         let branch = Configuration.LoadBranch()
-        let tagFilter = sprintf "fullbuild/%s/*" branch
-        match Tools.Vcs.FindLatestMatchingTag wsDir graph.MasterRepository tagFilter with
-        | Some tag -> let tagInfo = TagInfo.Parse tag
-                      Baseline(graph, tagInfo, false)
-        | _ -> let tagInfo = { TagInfo.BuildBranch = branch; TagInfo.BuildNumber = "temp" }
-               Baseline(graph, tagInfo, true)
-
-    member this.CreateBaseline (buildNumber : string) : Baseline =
-        let globals = Configuration.LoadGlobals()
-        let antho = Configuration.LoadAnthology()
-        let graph = Graph.from globals antho
+        let buildInfo = { BuildInfo.BuildBranch = branch; BuildInfo.BuildNumber = "1.0.0" }
+        Baseline(buildInfo, resultBaseline) 
+
+    member this.GetBinariesBaseline () : Baseline option =
```

Comment:
I can't see the difference between GetSourcesBaseline() and GetBinariesBaseline().

Moreover, GetBinariesBaseline() is not used...